danesfield/core/management/commands/populate_dev_data.py

```diff
@@ -19,20 +19,27 @@
 
 @click.command()
 def populate_dev_data():
-    point_cloud_file = ChecksumFile.objects.create(
-        name='TilesetWithDiscreteLOD.las',
-        file=ContentFile("i'm a point cloud (not really)").read(),
-    )
-    dataset: Dataset = Dataset.objects.create(
-        name='TilesetWithDiscreteLOD',
-        imageless=True,
-        point_cloud_file=point_cloud_file,
-    )
+    if Dataset.objects.filter(name='TilesetWithDiscreteLOD').exists():
+        dataset: Dataset = Dataset.objects.get(name='TilesetWithDiscreteLOD')
+        click.echo(f'Dataset "TilesetWithDiscreteLOD" already exists (pk = {dataset.pk}).')
+    else:
+        point_cloud_file = ChecksumFile.objects.create(
+            name='TilesetWithDiscreteLOD.las',
+            file=SimpleUploadedFile(name='TilesetWithDiscreteLOD.las', content=b'A' * 10),
+        )
+        dataset: Dataset = Dataset.objects.create(
+            name='TilesetWithDiscreteLOD',
+            imageless=True,
+            point_cloud_file=point_cloud_file,
+        )
```

Comment:
Does `Dataset.objects.get_or_create` not work here?

Reply:
No. Since there's a constraint on `Dataset` that prevents an imageless Dataset from being created without a `point_cloud_file`, you can't call `get_or_create` without supplying a `point_cloud_file`. But, there's no way to know whether or not the `Dataset` exists before calling `get_or_create`, so there's no way to know whether or not a `point_cloud_file` should be created or not.

In other words, to call `Dataset.objects.get_or_create(...)` you need a `point_cloud_file` already created. To create a `point_cloud_file`, you need to know if the `Dataset` exists or not (well, unless we didn't care about generating `point_cloud_files` every time the script is run, regardless of whether or not they're used).